File: challenge-load-test-chat/test-by-locust/src/tasks/list_messages.py
from locust import task, TaskSet
from endpoints import Endpoints

class ListMessagesTask(TaskSet):
    @task
    def list_messages(self):
        chat_id = getattr(self.user, 'chat_id', None)
        if chat_id:
            self.client.get(Endpoints.MESSAGE_LIST.format(chat_id=chat_id))

File: challenge-load-test-chat/test-by-k6/src/tasks/list-messages.ts
import http from 'k6/http';
import { check } from 'k6';
import { Endpoints } from '../endpoints';

export function listMessages(baseUrl: string, chatId: string) {
  const res = http.get(`${baseUrl}${Endpoints.MESSAGE_LIST(chatId)}`);
  check(res, { 'list messages status is 200': (r) => r.status === 200 });
  return res;
}
